crates/bin/pmonitor/src/main.rs

```diff
@@ -0,0 +1,465 @@
+                        let wallet_dir = self.wallet_path(&wallet_id);
+                        self.create_wallet(&wallet_dir, &new_fvk, &pmonitor_config.grpc_url())
+                            .await?;
+
+                        let new_fvk_entry = FvkEntry {
+                            fvk: new_fvk.clone(),
+                            wallet_id,
+                        };
+                        // Mark that the config needs to get saved again for the next time we run the audit command.
+                        config_updated = true;
+
+                        // We need to update the config with the new FVK and path on disk
+                        // to the wallet for the next time we run the audit command.
+                        let mut new_config_entry = config.clone();
+                        new_config_entry.add_migration(new_fvk_entry);
+                        updated_config.set_account(index, new_config_entry.clone());
+
+                        let mut view_client = self
+                            .view(wallet_dir, new_fvk.clone(), pmonitor_config.grpc_url())
+                            .await?;
+
+                        println!("Syncing new wallet...");
+                        self.sync(&mut view_client).await?;
+                        println!("Wallet synced successfully");
+                        // Now we can exit the else if statement and continue by computing the balance,
+                        // which will use the new migrated wallet.
+                    } else {
+                        // we expect a single migration tx per FVK, if this assumption is violated we should bail.
+                        anyhow::bail!(
+                            "Expected a single migration tx, found {}",
+                            migration_tx.len()
+                        );
+                    }
+
+                    let total_um_equivalent_amount = self
+                        .compute_um_equivalent_balance(&mut view_client, &mut stake_client)
+                        .await?;
+
+                    println!("Original FVK: {:?}", config.original_fvk());
+                    let genesis_um_equivalent_amount = config.genesis_balance();
+                    println!(
+                        "Genesis UM-equivalent balance: {:?}",
+                        genesis_um_equivalent_amount
+                    );
+                    println!(
+                        "Current UM-equivalent balance: {:?}",
+                        total_um_equivalent_amount
+                    );
+
+                    // Let the user know if the balance is unexpected or not
+                    if total_um_equivalent_amount < genesis_um_equivalent_amount {
+                        println!("✘ Unexpected balance! Balance is less than the genesis balance");
+                    } else {
+                        println!("✅ Expected balance! Balance is greater than or equal to the genesis balance");
```

Comment:
Love the emojis, makes the wall of text a bit easier to scan visually. I only tested with 10 FVKs, though, and most of the relevant content was off the screen. Maybe `tracing` statements would be more appropriate here.